core/include/fitter/kalman_fitter.hpp

```diff
@@ -38,11 +44,91 @@ class kalman_fitter {
         /// Whether run reversed filtering
         bool reversed_filtering = false;
 
+	actor(int target_surface_id,
+	      traccc::host_measurement_collection& input_measurements,
+	      traccc::host_surface_collection& surfaces,
+	      bool multiple_scattering = true,
+	      bool energy_loss = true,
+	      bool reversed_filtering = false) :
+	    target_surface_id(target_surface_id),
+	    input_measurements(input_measurements),
+	    surfaces(surfaces),
+	    multiple_scattering(multiple_scattering),
+	    energy_loss(energy_loss),
+	    reversed_filtering(reversed_filtering) {}
+
+
         template <typename propagator_state_t, typename stepper_t>
         void operator()(propagator_state_t& state,
-                        const stepper_t& stepper) const {}
+                        const stepper_t& stepper) const {
+
+	    // + Find the surface
+	    if (target_surface_id >= surfaces.items.size())
+		return;
+	    traccc::surface &surf = surfaces.items.at(target_surface_id);
+
+	    // Find matching measurement
+	    auto res = std::find_if(input_measurements.begin(), input_measurements.end(),
+			 [this] (traccc::measurement &m) {
+			     return m.surface_id == target_surface_id;
+			 }
+	    );
+	    if (res == input_measurements.end())
+		return;
+	    traccc::measurement meas = *res;
+
+	    // + Transport covariance to the surface
+	    stepper.cov_transport(state);
+	    
+	    // + Pre Material effects (ignored for now)
+	    
+	    // + Bind stepping state to the surface, get boundParams, jacobian, pathlength
+	    // Note: this should be in the stepper
+	    auto [boundParams, jacobian, pathLength] =
+		traccc::detail::bound_state(
+		    state.stepping.cov,
+		    state.stepping.jacobian,
+		    state.stepping.jac_transport,
+		    state.stepping.derivative,
+		    state.stepping.jac_to_global,
+		    state.stepping.pars,
+		    // state.stepping.covTransport,
+		    false, // FIXME!!!!
```

Comment:
@beomki-yeo can you comment on this? If I set this flag to true (that's covTransport), the update does not work. Is this expected?